app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -360,79 +402,79 @@ private void takeAPhoto(){
         startActivityForResult(intent, TAKE_PHOTO);
     }
 
-    private byte[] compress(Bitmap photo){
+    private byte[] compress(Bitmap photo) {
         int quality = 100;
 
-        if (photo.getByteCount()<= MAX_PHOTO_SIZE) {
+        if (photo.getByteCount() <= MAX_PHOTO_SIZE) {
             photo.compress(Bitmap.CompressFormat.PNG, quality, photoStream);
             return photoStream.toByteArray();
         }
 
         //
-        while(true) {
+        while (true) {
             photo.compress(Bitmap.CompressFormat.JPEG, quality, photoStream);
-            byte [] compressedPhoto = photoStream.toByteArray();
-            if(compressedPhoto.length <= MAX_PHOTO_SIZE || quality == 5){
+            byte[] compressedPhoto = photoStream.toByteArray();
+            if (compressedPhoto.length <= MAX_PHOTO_SIZE || quality == 5) {
                 return compressedPhoto;
-            }
-            else {
+            } else {
                 quality -= 5;
             }
         }
     }
 
-    protected void onActivityResult(int requestCode, int resultCode, Intent intent){
+    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
         Log.d("tag", "In Take Photo");
         TextView modePicture = (TextView) findViewById(R.id.modePhoto);
-        if (requestCode == TAKE_PHOTO){
+        if (requestCode == TAKE_PHOTO) {
 
-            if (resultCode == RESULT_OK){
+            if (resultCode == RESULT_OK) {
                 Log.d("tag", "Taking photo");
 
                 Bitmap photo = (Bitmap) intent.getExtras().get("data");
 
+                // Clear the old photoStream
+                photoStream.reset();
+
                 // Compress the photo if needed
-                byte [] compressedPhoto = compress(photo);
+                byte[] compressedPhoto = compress(photo);
 
                 // Encode photo to string here
+                encodedPhoto = ""; //Clear the old String just in case.
                 encodedPhoto = Base64.encodeToString(compressedPhoto, Base64.DEFAULT);
 
                 // This is how to decode the photo
-                                                    // Give it a String encoded[Photo/Emoji]
+                // Give it a String encoded[Photo/Emoji]
                 byte[] decodedString = Base64.decode(encodedPhoto, Base64.DEFAULT);
                 Bitmap decodedPhoto = BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);
 
                 // Set the image here
-                ImageView imageView = (ImageView)findViewById(R.id.imgView);
-                imageView.setImageBitmap(photo);
+                ImageView imageView = (ImageView) findViewById(R.id.imgView);
+                imageView.setImageBitmap(decodedPhoto);
```

Comment:
Although we could just load the original photo, loading the decodedPhoto after compression/encoding allows us to visually see whether or not our photo handling process worked. As such, I feel this is alright to leave in, even if it might require an extra 10ns to load.